explainaboard/metrics/metric.py

```diff
@@ -371,16 +371,39 @@ def get_batch_data(self) -> np.ndarray[tuple[int, int, int], Any]:
 
 
 class Metric(metaclass=abc.ABCMeta):
-    """A class representing an evaluation metric."""
+    """A class representing an evaluation metric.
+
+    When a subclass needs to construct a random number generator, initialize the random
+    generator as follows: Invoke the `get_seed()` method to get a numpy SeedSequence,
+    spawn the SeedSequence with the SeedSequence's `spawn` method, and then pass the
+    spawned SeedSequence to the random generator's constructor.
+
+    Example:
+        class FooMetric(Metric):
+            def foo(self) -> None:
+                # Spawns a SeedSequence.
+                seed = self.get_seed().spawn(1)[0]
+                # Initializes a random generator with the spawned seed.
+                rng = np.random.default_rng(seed)
+                # Do something with `rng`.
+
+        config = MetricConfig(...)
+        metric = FooMetric(config, seed=12345)
+        metric.foo()
+    """
 
     config: MetricConfig
 
-    def __init__(self, config: MetricConfig):
+    def __init__(self, config: MetricConfig, seed: int | None = None):
```

Comment:
Give SeedSequence rather than int so that this class can be constructed by other classes that may hold their own SeedSequence:
```python
class Routine:
  def __init__(self, seed: np.SeedSequence | None):
    self._seed = seed if seed is not None else np.SeedSequence()
    self._inner_routine = InnerRoutine(seed=self._seed.spawn())

  def do_something(self):
    self._inner_routine.do_something_else()
```

Reply:
Changed to take `np.random.SeedSequence`.